test/cql-pytest/test_group_by.py

```diff
@@ -38,13 +38,47 @@ def test_group_by_partition(cql, table1):
 
 # Adding a LIMIT should be honored.
 # Reproduces #17237 - more results than the limit were generated.
-@pytest.mark.xfail(reason="issue #17237")
 def test_group_by_partition_with_limit(cql, table1):
     # "LIMIT 1" should return only one of the two matches (0,0,0,1) and
     # (1,0,0,5). The partition key 1 happens to be the first one in
     # murmur3 order, so this is what should be returned.
     assert {(1,0,0,5)} == set(cql.execute(f'SELECT p,c1,c2,v FROM {table1} GROUP BY p LIMIT 1'))
 
+def test_group_by_partition_count_with_limit(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "a int, b int, c int, PRIMARY KEY (a, b)") as table:
+        # Insert 4 rows with the same partition key, but different clustering keys
+        # 29, 31, 37, 41 are primes, so they have no common factors with each
+        # other, and will not randomly become a result of multiplication.
+        for i in range(29):
+            cql.execute(f"insert into {table} (a,b,c) values ({29},{i+1},29);");
+        for i in range(31):
+            cql.execute(f"insert into {table} (a,b,c) values ({31},{i+1},31);");
+        for i in range(37):
+            cql.execute(f"insert into {table} (a,b,c) values ({37},{i+1},37);");
+        for i in range(41):
+            cql.execute(f"insert into {table} (a,b,c) values ({41},{i+1},41);");
+        query_result = list(cql.execute(f'SELECT a FROM {table} GROUP BY a'))
+        # We should get 4 rows, one for each partition key
+        assert len(query_result) == 4
+        assert {29, 31, 37, 41} == set(map(lambda x: x[0], query_result))
+        # Adding a LIMIT should be honored.
+        for i in range(1,4):
+            query_result = set(map(lambda x: x[0], cql.execute(f'SELECT a FROM {table} GROUP BY a LIMIT {i}')))
+            # We should get only {i} rows, one of the partition keys
+            assert len(query_result) == i
+            assert query_result.issubset({29, 31, 37, 41})
+        # COUNT should return proper count values (e.g. 29 for partition key 29)
+        query_result = list(cql.execute(f'SELECT COUNT(a) FROM {table} GROUP BY a'))
+        # We should get 4 rows, one for each partition key
+        assert len(query_result) == 4
+        assert {29, 31, 37, 41} == set(map(lambda x: x[0], query_result))
```

Comment:
The more pythonic (and nicer) syntax of this lambda thing is "list comprehension":

```python
assert {29, 31, 37, 41} == {x[0] for x in query_result}
```